### ISSUE TITLE
Rename Symbol skips unreachable code

### ISSUE THREAD

**Issue by xavdid** [microsoft/pylance-release#2431]
<h3> Environment data </h3>

-   Language Server version: 2022.2.1
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: basic

<h3> Expected behaviour </h3>

All references to the same symbol are renamed, even those that are otherwise unreachable.

<h3> Actual behaviour </h3>

Unreachable symbols aren't renamed. If the line causing code to be unreachable is removed after rename, then you have undefined variables caused by the renamer.

Apologies in advance if this is working as intended, it just seemed like odd behavior to me.

<h3> Logs </h3>

Python Language Server Log
```
(there was no data on my clipboard); manually fetched logs didn't seem relevant. Happy to include them if needed.
```

<h3> Code Snippet / Additional information </h3>

Renaming `a` -> `b` doesn't rename unreachable symbol

![](https://cdn.zappy.app/456abe048fbf07731aea49ed4e964d97.gif)

```python
def f():
    a = 1
    a += 2
    return a

    a += 4 # isn't renamed
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.3.1, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202231-9-March-2022)
